MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -1310,6 +1463,36 @@ private static String adjustWeightsForFaction(String weights, String faction) {
 
         return unitTypes;
     }
+    
+    /**
+     * Specialized logic for generating clan units
+     * @return
+     */
+    private static List<Integer> generateClanUnitTypes(int unitCount, int forceQuality, String factionCode, Campaign campaign) {        
+        // per AtB, we have a 1/6 odds of encountering a vehicle star
+        // for fluff reasons, hell's horses + pals use more vehicles
+        // higher-rated clan units become increasingly unlikely to use vehicles
+        int vehicleTarget = 6;
+        if(factionCode.equals("CHH") || factionCode.equals("CSL") || factionCode.equals("CBS")) {
+            vehicleTarget = 8;
+        } else {
+            vehicleTarget -= forceQuality;
+        }
+        
+        // we randomly determine tank or mek
+        int roll = Compute.d6(2); 
```

Comment:
I'm pretty sure the logic or description here is flawed. A base target of 6 on 2d6 is 27.8% (just below 2/6) whereas a base target of 4 would be 16.7% (1/6) on 2d6.

Reply:
Unit quality theoretically ranges from 0 (F) to 5 (A*). For clans, that translates into "backwater Solamha" to "front line elite". The idea being that front line elites are too good for tanks (6 - 5 = 1 which is never on a 2d6) while dusty old warriors in the backwoods take what they can get (6 - 0 = 6+ which is sometimes). It's kind of a spitball thing. If it winds up not looking right, I'll adjust the odds going forward.

Anyway, I'll update the description.